src/frontend/src/binder/relation/table_or_source.rs

```diff
@@ -315,6 +315,55 @@ impl Binder {
         })
     }
 
+    pub(crate) fn get_table_by_name(
```

Comment:
Added this function so that we could look up table information (Table ID, Table Indexes, etc) without binding the table to the query context, because binding a table to the query context when it's not actually being used _in_ the query would create a lot of risk for unexpected behavior and bugs.

Reply:
Returning a `BoundBaseTable` that is not bound seems a bit werid. IIUC and as you mentioned, this method is added only to fetch the table id and table indexes of a table name. Given that we already have intial support on [regclass](https://github.com/risingwavelabs/risingwave/blob/351b1bca8e2a1269f91c08f069f9b1c559e56fe1/src/frontend/src/binder/expr/mod.rs#L419), I wonder whether we can leverage `regclass` and improve it instead of adding this method. Basically we can rewrite `pg_table_size(<param>)` to 
```
SELECT total_key_size + total_value_size
FROM rw_catalog.rw_table_stats t
WHERE t.id = <param>::regclass
```

The following changes is needed to `regclass` support:
- We need to support int literal. This is straightforward since we can just pass through.
- We need to support parsing the [class_name](https://github.com/risingwavelabs/risingwave/blob/351b1bca8e2a1269f91c08f069f9b1c559e56fe1/src/frontend/src/binder/expr/mod.rs#L442) in case it is a schema quailified name, which you already did in this PR.

Similar things apply to `pg_index_size`: you can get table id via regclass and get the indexes of a table via catalog.

Reply:
Agreed, the naming of `BoundBaseTable` would be poor design in the final code, in fact I had a TODO comment further below to remind me to call this out for feedback when I moved this from a Draft PR.

Using `regclass` would be much better,. I'll update the code to use that instead. It should be pretty easy to do.

Reply:
@hzxa21 I'm working on updating the code to use the `regclass` operation and I can't figure out the appropriate way to construct `'mytable'::regclass` from `ExprImpl` values.

I'm starting from the following but I cannot figure out how to properly provide operands to the `Cast` function such that the function inference logic picks the proper cast implementation (I'm assuming that `'mytable'::regclass` is compiled to an `ExprImpl::Cast`):

```rust
        FunctionCall::new_unchecked(
            ExprType::Cast,
            vec![
                table_name,
            ],
          /* what is the return type, there isn't a DataType::Regclass */
        )?;
```
> All cases in code where `ExprImpl::Cast` occured  `FunctionCall::new_unchecked` was used.

I was able to prove that this would work by using the more brute-force method below, but I don't think this is the right way to write this code:
```rust
self.bind_cast(
            Expr::Value(Value::SingleQuotedString(table_name)),
            AstDataType::Regclass,
        )?
```

Reply:
> I don't think this is the right way to write this code:

```rust
self.bind_cast(
    Expr::Value(Value::SingleQuotedString(table_name)),
    AstDataType::Regclass,
)?
```

What's wrong with this approach? I think it's acceptable. `bind_cast` should be stateless. 🤔

Reply:
ping @hzxa21

Reply:
> > I don't think this is the right way to write this code:
> 
> ```rust
> self.bind_cast(
>     Expr::Value(Value::SingleQuotedString(table_name)),
>     AstDataType::Regclass,
> )?
> ```
> 
> What's wrong with this approach? I think it's acceptable. `bind_cast` should be stateless. thinking

Wasn't sure if using `AstDataType` in this code when everything else uses `DataType` was an issue.  Seemed like mixing two different systems.

Reply:
As you have pointed out, `AstDataType` belongs to the parser, while `DataType` is what the rest of RisingWave works with. There is only `AstDataType::Regclass` but no correspondence in `DataType`.

This is due to the fact that `regclass` / `oid` are not first class citizens of RisingWave's type system at this time. As the component that transforms `AstFoo` to `BoundFoo` (incl. `AstDataType` to `DataType`), the binder offers limited support to `regclass` / `oid` to facilitate the pg_foo tables / views / functions. Following this practice, it is acceptable to interact with both `AstDataType` and `DataType` in the binder, although it is not supposed to construct new Ast nodes unless immediately consumed by itself and there are no better options.

As for your concern about the different between `oid` and `regclass`. This is what [PostgreSQL](https://www.postgresql.org/docs/current/datatype-oid.html) says:

> The OID alias types have no operations of their own except for specialized input and output routines. These routines are able to accept and display symbolic names for system objects, rather than the raw numeric value that type `oid` would use. The alias types allow simplified lookup of OID values for objects.

That is, they are same except for parsing and display format.